sqlglot/parser.py

```diff
@@ -4090,16 +4090,14 @@ def _parse_interval(self, match_interval: bool = True) -> t.Optional[exp.Add | e
         if this and this.is_number:
             this = exp.Literal.string(this.name)
         elif this and this.is_string:
-            parts = this.name.split()
-
-            if len(parts) == 2:
+            parts = exp.INTERVAL_STRING_RE.findall(this.name)
+            if len(parts) == 1:
```

Comment:
Is this branch condition matching the previous one? 

Seems like `parts` is now a 2-D array, maybe the transferred condition should be `len(parts[0]) == 2` ?

Reply:
Nope, if you had `1 day` it'd previously be split into `[1, 'day']`. Now it's split into [(1, 'day')]`. Different representation.